test/harness/e2e/harness.go

```diff
@@ -177,12 +179,43 @@ func (h *Harness) GetDeviceWithStatusSystem(enrollmentID string) *apiclient.Read
 	return device
 }
 
+func (h *Harness) GetDeviceWithStatusSummary(enrollmentID string) v1alpha1.DeviceSummaryStatusType {
+	device, err := h.Client.ReadDeviceWithResponse(h.Context, enrollmentID)
+	Expect(err).NotTo(HaveOccurred())
+	// we keep waiting for a 200 response, with filled in Status.SystemInfo
+	if device.JSON200 == nil || device.JSON200.Status == nil || device.JSON200.Status.Summary.Status == "" {
+		return ""
+	}
+	return device.JSON200.Status.Summary.Status
+}
+
+func (h *Harness) PatchDeviceWithBody(deviceID string, body string) *apiclient.PatchDeviceResponse {
+	buf, err := json.Marshal(body)
+	if err != nil {
+		return nil
+	}
+	resp, err := h.Client.PatchDeviceWithBodyWithResponse(h.Context, deviceID, "application/json", bytes.NewReader(buf))
+	Expect(err).NotTo(HaveOccurred())
+	if resp != nil && resp.JSON200 != nil {
+		return nil
+	}
+	return resp
+}
+
 func (h *Harness) ApiEndpoint() string {
 	ep := os.Getenv("API_ENDPOINT")
 	if ep == "" {
-		ep = "https://" + util.GetExtIP() + ":3443"
-		logrus.Infof("API_ENDPOINT not set, using default: %s", ep)
-		err := os.Setenv("API_ENDPOINT", ep)
+		logrus.Error("API_ENDPOINT must be set")
+	}
+	return ep
+}
+
+func (h *Harness) RegistryEndpoint() string {
+	ep := os.Getenv("REGISTRY_ENDPOINT")
+	if ep == "" {
+		ep = "https://" + util.GetExtIP() + ":5000"
```

Comment:
```suggestion
		ep = "https://" + util.GetExtIP() + ":5000"
```

Do the same here as for the other, GetExtIP isn't there anymore :)

Reply:
Done, thanks